client-misk-jooq/src/test/kotlin/app/cash/backfila/client/misk/setup/JooqWidgetCompoundKeyBackfill.kt

```diff
@@ -0,0 +1,139 @@
+package app.cash.backfila.client.misk.setup
+
+import app.cash.backfila.client.misk.BackfillConfig
+import app.cash.backfila.client.misk.Description
+import app.cash.backfila.client.misk.jooq.BackfillBatch
+import app.cash.backfila.client.misk.jooq.BackfillJooqTransacter
+import app.cash.backfila.client.misk.jooq.ByteStringSerializer
+import app.cash.backfila.client.misk.jooq.ByteStringSerializer.Companion.forByteArray
+import app.cash.backfila.client.misk.jooq.ByteStringSerializer.Companion.forString
+import app.cash.backfila.client.misk.jooq.JooqBackfill
+import app.cash.backfila.client.misk.jooq.gen.tables.records.WidgetsRecord
+import app.cash.backfila.client.misk.jooq.gen.tables.references.WIDGETS
+import com.google.common.base.CharMatcher
+import org.jooq.Condition
+import org.jooq.Field
+import org.jooq.Record
+import org.jooq.TableLike
+import javax.inject.Inject
+
+data class CompoundKey(
+  val manufacturerToken: String,
+  val createdAtMs: Long,
+  val widgetToken: ByteArray
+) {
+  fun toRecord(): Record {
+    val record = WidgetsRecord()
+    record.manufacturerToken = manufacturerToken
+    record.createdAtMs = createdAtMs
+    record.widgetToken = widgetToken
+    return record
+  }
+
+  fun encode(): String {
+    return "$manufacturerToken:" +
+      "${zeroPad(createdAtMs, 16)}:" +
+      forString.fromByteString(forByteArray.toByteString(widgetToken))
+  }
+
+  private fun zeroPad(number: Long, width: Int): String {
+    return String.format("%0" + width + "d", number)
+  }
+
+  override fun equals(other: Any?): Boolean {
```

Comment:
Since this is kotlin and is a data class we don't need to override equals, hashcode or tostring right?

Reply:
actually i was wondering that too, but the compiler throws warning because there's a ByteArray in there.

Reply:
Yeah ok, could you add a small comment to that effect?